app/assets/javascripts/analytics-ga4/ga4-finder-tracker.js

```diff
@@ -0,0 +1,140 @@
+        filterSections[i].setAttribute('data-ga4-index', JSON.stringify({ index_section: i + 1, index_section_count: filterSections.length }))
+      }
+    },
+
+    // changeEventMetadata is a string referring to the type of form change and the element type that triggered it. For example 'update-filter checkbox'.
+    trackChangeEvent: function (eventTarget, changeEventMetadata) {
+      var PIIRemover = new window.GOVUK.analyticsGa4.PIIRemover()
+
+      changeEventMetadata = changeEventMetadata.split(' ')
+
+      var elementType = changeEventMetadata[1]
+
+      var section = eventTarget.closest('[data-ga4-section]')
+
+      var schema = new GOVUK.analyticsGa4.Schemas().eventSchema()
+      schema.event = 'event_data'
+      schema.event_data.type = 'finder'
+
+      var elementInfo = this.getElementInfo(eventTarget, elementType, section)
+      var elementValue = elementInfo.elementValue
+      var wasFilterRemoved = elementInfo.wasFilterRemoved
+
+      schema.event_data.text = elementValue
+
+      var changeType = changeEventMetadata[0]
+      switch (changeType) {
+        case 'update-filter':
+          schema.event_data.event_name = 'select_content'
+
+          if (section) {
+            schema.event_data.section = section.getAttribute('data-ga4-section')
+          }
+
+          if (wasFilterRemoved) {
+            schema.event_data.action = 'remove'
+            schema.event_data.text = elementType === 'text' ? undefined : elementValue
+          } else {
+            schema.event_data.action = elementType === 'text' ? 'search' : 'select'
+            schema.event_data.index = this.getSectionIndex(section)
+          }
+          break
+
+        case 'update-keyword':
+          schema.event_data.event_name = 'search'
+          schema.event_data.url = window.location.pathname
+          schema.event_data.section = 'Search'
+          schema.event_data.action = 'search'
+          schema.event_data.text = PIIRemover.stripPIIWithOverride(schema.event_data.text, true, true)
+          break
+
+        case 'clear-all-filters':
+          schema.event_data.event_name = 'select_content'
+          schema.event_data.action = 'remove'
+          schema.event_data.text = 'Clear all filters'
+          break
+
+        case 'update-sort':
+          schema.event_data.event_name = 'select_content'
+          schema.event_data.action = 'sort'
+          schema.event_data.section = 'Sort by'
+          break
+      }
+
+      window.GOVUK.analyticsGa4.core.sendData(schema)
+    },
+
+    getElementInfo: function (eventTarget, elementType, section) {
+      var elementValue = ''
+      var defaultValue
+      var wasFilterRemoved = false
+      if (elementType === 'checkbox') {
+        var checkboxId = eventTarget.id
+
+        // The "value" we need for a checkbox is the label text that the user sees beside the checkbox.
+        elementValue = document.querySelector("label[for='" + checkboxId + "']").textContent
```

Comment:
I think you could combine lines 91 and 94?

```Javascript
elementValue = document.querySelector("label[for='" + eventTarget.id + "']").textContent
```

Reply:
(same for radios)

Reply:
Cool, I believe my thinking was it would be more readable having the named variable there, but I have changed it to your suggestion. :+1: